### PR TITLE
xdsl: make flake8 happy with our source files

### PR DESCRIPTION
**Pull request by superlopuh**
I ran `python -m flake8 xdsl`, and went about getting the number of lint messages to 0.

I recommend reviewing by commit, and if there are any changes you're not happy with, let's take them out to a separate PR because these kinds of things tend to diverge from everything else people are working on pretty quickly, I'd love to merge as much of this as possible soon

### CODE REVIEW

File: xdsl/dialects/builtin.py [hunk 1]
```diff
@@ -579,11 +579,6 @@ class Float16Type(ParametrizedAttribute, MLIRType):
 f16 = Float16Type()
 
 
-@irdl_attr_definition
```
Comment (webmiche) [inline xdsl/dialects/builtin.py:582]: This is actually something that is needed. I feel that the proper way to go about this not being used is adding a testcase.

**Review by webmiche — APPROVED** [review]
LGTM, module one comment. We can move this out of this PR into an issue, if you want, or, if you have the time, you could add the test.

**Review by webmiche — CHANGES_REQUESTED** [review]
Sorry, missclicked :sweat_smile:

File: xdsl/dialects/builtin.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/builtin.py:582]: This is duplicate code of lines 140-142 :)

Comment (webmiche) [reply xdsl/dialects/builtin.py:582]: Ah sure, then remove it :+1:

**Review by webmiche — APPROVED** [review]

**Review by math-fehr — APPROVED** [review]
Thanks a lot!
After the paper deadline, I'll look more into adding this to the CI!

**Pull request merged**